chainstate/src/interface/chainstate_interface.rs

```diff
@@ -166,4 +166,31 @@ pub trait ChainstateInterface: Send {
 
     /// Returns true if the initial block download isn't finished yet.
     fn is_initial_block_download(&self) -> Result<bool, ChainstateError>;
+
+    fn pool_exists(&self, pool_id: PoolId) -> Result<bool, ChainstateError>;
+
+    fn get_pool_balance(&self, pool_id: PoolId) -> Result<Option<Amount>, ChainstateError>;
```

Comment:
What if we make `PoSAccountingView` a  supertrait of `ChainstateInterface`?

Reply:
I am open to this possibility. However, as @TheQuantumPhysicist noted above, chainstate interface already contains many methods so it would be better to be more explicit with naming, e.g. `pool_exists` -> `stake_pool_exists`. In that case, the `PoSAccountingView` would have to be renamed too. Keeping them separate allows us to keep the more concise names that go with the more specialized interface.

Ultimatley I don't have a strong preference either way. If we add `PoSAccountingView` as a supertrait, we should consider whether the same should be done with `UtxosView` as well.